jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/LogEventHandler.java

```diff
@@ -17,12 +17,19 @@
 package com.google.cloud.tools.jib.gradle;
 
 import com.google.cloud.tools.jib.event.events.LogEvent;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 import java.util.function.Consumer;
 import org.gradle.api.logging.Logger;
 
 /** Handles {@link LogEvent}s by passing to the Gradle {@link Logger}. */
+// We don't care about the return values of the logging futures.
+@SuppressWarnings("FutureReturnValueIgnored")
 class LogEventHandler implements Consumer<LogEvent> {
 
+  /** This executor keeps all log messages in order. */
+  private static final ExecutorService executorService = Executors.newSingleThreadExecutor();
```

Comment:
Hmm... this will do serialize the log messages in sequence, but I think this will make the logs out of sync with the actual event points, since logs are being handled in a new, separate thread.

I'm still wondering why this is happening all of a sudden with Gradle, because in the end, the code seems to do the same thing as before.

Reply:
This apparently happened a long time ago, and the same fix needed to be applied to `GradleJibLogger`. The same fix was just missing from the event handler. @coollog can probably explain better.

Reply:
Yeah, this was because calling the Gradle logging from different threads may cause the log messages to be out-of-order. I think having the log messages in sync with the actual event points is not too important as long as we have them in the same order in which they are emitted (ensured via the single-thread executor).

Reply:
Okay, this is not a regression, so we don't have to be concerned in this PR.

I think we can just file an issue about the out-of-sync log messages in Gradle to keep track of it.